deployment/registry.py

```diff
@@ -1,36 +1,43 @@
 import json
+from collections import defaultdict
 from enum import Enum
 from pathlib import Path
-from typing import Dict, List, NamedTuple, Optional
+from typing import Dict, List, NamedTuple, Optional, Union
 
 from ape.contracts import ContractInstance
 from eth_typing import ChecksumAddress
 from eth_utils import to_checksum_address
+from web3.types import ABI
 
 from deployment.params import get_contract_container
 from deployment.utils import check_registry_filepath
 
 
-class RegistryEntry(NamedTuple):
-    """Represents a single entry in a nucypher-style contract registry."""
+ChainId = int
+ContractName = str
+ContractArtifacts = Dict[ContractName, Union[ChecksumAddress, ABI]]
+RegistryData = Dict[ChainId, ContractArtifacts]
+
 
-    contract_name: str
-    contract_version: str  # TODO: remove version from registry
-    contract_address: ChecksumAddress
-    contract_abi: List[Dict]
+class ContractEntry(NamedTuple):
```

Comment:
What's the reason we use two different representations for the same information i.e. `RegistryData` and `ContractEntry`

Reply:
`ContractEntry` is a single contract entry in the registry abstracted into a `NamedTuple` used for internal processing.  `RegistryData` is the entire registry data structure as a `Dict` type alias for type annotations.

Reply:
I didn't express this very well. 

Basically, `RegistryData` only applies to writing to the registry, vs `ContractEntry`, which is used everywhere else. 

Perhaps `write_registry(...)` should take List[ContractEntry] as the parameter, and generate the `RegistryData` internally instead of having callers of the function (`merge_registry`, `registry_from_ape_deployments`) do the conversion to RegistryData before calling. That way there is only place doing the conversion to `RegistryData`...and you barely even really need a separate type because it is no longer a parameter.

Reply:
Nice suggestions ✔️